src/commonMain/kotlin/fr/acinq/eclair/channel/Channel.kt

```diff
@@ -2631,19 +2672,20 @@ data class Closing(
                             commitments.payments[add.id]?.let { paymentId -> logger.info { "c:$channelId paymentId=$paymentId will settle on-chain (htlc #${add.id} sending ${add.amountMsat})" } }
                         }
 
-                        val nextState = when (val closingType = closing1.isClosed(watch.tx)) {
-                            null -> closing1
+                        val (nextState, closedActions) = when (val closingType = closing1.isClosed(watch.tx)) {
+                            null -> Pair(closing1, listOf<ChannelAction>())
```

Comment:
> IntelliJ should be showing a warning

It's not :frowning: IntelliJ has been horrible lately. It can't properly type-check the iOS specific files. It always struggles with SQLDelight files. And for several weeks type-checking of Gradle files was completely broken. And then it couldn't find `kolinx` for several weeks, which resulted in red highlights everywhere.

Reply:
Damn that must be frustrating...good luck :sweat_smile: